src/java.base/share/classes/jdk/internal/module/ModuleBootstrap.java

```diff
@@ -469,15 +468,27 @@ private static ModuleLayer boot2() {
                 limitedFinder = new SafeModuleFinder(finder);
         }
 
+        // If -Xshare:dump and mainModule are specified, check if the mainModule
+        // is in the runtime image and not in the upgrade module path. If so,
```

Comment:
Small typo here, should be "on the upgrade module path" rather than "in".

Reply:
Fixed